src/js/blocks/hero/block.js

```diff
@@ -0,0 +1,379 @@
+const { __ } = wp.i18n;
+const { registerBlockType } = wp.blocks;
+const {
+	Dashicon,
+	IconButton,
+	PanelBody,
+	PanelRow,
+	Toolbar,
+	withNotices,
+	BaseControl,
+	TextControl,
+	Button,
+	ButtonGroup,
+	ToggleControl
+} = wp.components;
+const {
+	BlockControls,
+	InspectorControls,
+	BlockAlignmentToolbar,
+	MediaPlaceholder,
+	MediaUpload,
+	MediaUploadCheck,
+	AlignmentToolbar,
+	RichText,
+	PlainText,
+	URLInput
+} = wp.blockEditor;
+
+// @see https://github.com/WordPress/gutenberg/tree/master/packages/block-library/src
+
+const ALLOWED_MEDIA_TYPES = [ 'image' ];
+
+const customIcon = () => {
+	return (
+		<img
+			width="20"
+			height="20"
+			className="dashicon"
+			src={ ( URI_CL_URL + 'i/icons/hero.png' ) }
+			alt="button"
+		/>
+	);
+};
+
+const randomID = () => {
+
+	// https://stackoverflow.com/questions/6860853/generate-random-string-for-div-id
+	let S4 = () => {
+		return ( ( ( 1 + Math.random() ) * 0x10000 ) | 0 ).toString( 16 ).substring( 1 );
+	};
+	return ( S4() + S4() + '-' + S4() + '-' + S4() + '-' + S4() + '-' + S4() + S4() + S4() );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor. (new-cap)](https://app.codacy.com/manual/uriweb/uri-component-library/pullRequest?prid=4757242)